src/zkevm_specs/evm_circuit/execution/precompiles/ecadd.py

```diff
@@ -0,0 +1,50 @@
+from zkevm_specs.evm_circuit.instruction import Instruction
+from zkevm_specs.evm_circuit.table import (
+    CallContextFieldTag,
+    EccOpTag,
+    FixedTableTag,
+    RW,
+)
+from zkevm_specs.util import FQ, Word, Bn254AddGas
+
+
+def ecAdd(instruction: Instruction):
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess, RW.Read)
+    address_word = instruction.call_context_lookup_word(CallContextFieldTag.CalleeAddress)
+    address = instruction.word_to_address(address_word)
+    instruction.fixed_lookup(
+        FixedTableTag.PrecompileInfo,
+        FQ(instruction.curr.execution_state),
+        address,
+        FQ(Bn254AddGas),
+    )
+
+    # Get p, q and out from aux_data
+    px: Word = instruction.curr.aux_data[0]
+    py: Word = instruction.curr.aux_data[1]
+    qx: Word = instruction.curr.aux_data[2]
+    qy: Word = instruction.curr.aux_data[3]
+    outx: FQ = instruction.curr.aux_data[4]
+    outy: FQ = instruction.curr.aux_data[5]
+
+    # if is_success is false, outx and outy are zero
+    if is_success == FQ.zero():
+        instruction.constrain_zero(outx)
+        instruction.constrain_zero(outy)
+
+    # ecc table lookup
+    instruction.ecc_lookup(FQ(EccOpTag.Add), px, py, qx, qy, FQ.zero(), outx, outy, is_success)
+
+    # consume gas
```

Comment:
duplicated?

Reply:
remvoed in [b6c812d](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/500/commits/b6c812d260d279ec95bbd8f9525d58dcbce75559)